processing/src/main/java/io/druid/segment/incremental/OnheapIncrementalIndex.java

```diff
@@ -366,14 +390,24 @@ public void close()
   // here its made thread safe to support the special case of groupBy where the multiple threads can add concurrently to the IncrementalIndex.
   static class ObjectCachingColumnSelectorFactory implements ColumnSelectorFactory
   {
-    private final ConcurrentMap<String, LongColumnSelector> longColumnSelectorMap = Maps.newConcurrentMap();
-    private final ConcurrentMap<String, FloatColumnSelector> floatColumnSelectorMap = Maps.newConcurrentMap();
-    private final ConcurrentMap<String, ObjectColumnSelector> objectColumnSelectorMap = Maps.newConcurrentMap();
+    private final Map<String, LongColumnSelector> longColumnSelectorMap;
+    private final Map<String, FloatColumnSelector> floatColumnSelectorMap;
+    private final Map<String, ObjectColumnSelector> objectColumnSelectorMap;
     private final ColumnSelectorFactory delegate;
 
-    public ObjectCachingColumnSelectorFactory(ColumnSelectorFactory delegate)
+    public ObjectCachingColumnSelectorFactory(ColumnSelectorFactory delegate, boolean concurrentEventAdd)
     {
       this.delegate = delegate;
+
+      if (concurrentEventAdd) {
+        longColumnSelectorMap = Maps.newConcurrentMap();
```

Comment:
Please use `ConcurrentHashMap`, the implementation behind `Maps.newConcurrentMap()` has higher overhead and is less scalable.